### PR TITLE
[SPARK]Use timestamp with zone in iceberg format

### LINKED ISSUE #686: [Bug][Spark]: Spark write timestamp value to parquet as UTC+0 hour.
### What happened?

When table schema contains a field as timestamp without-zone,  the value spark engines write to the parquet file is UTC+0, but should be the current timezone value.

### Affects Versions

0.4.0

### What engines are you seeing the problem on?

Spark

### How to reproduce

checked .

select via flink 

image

select via spark 

image

table schema 

image

table files 

image

seems spark problems

### Relevant log output

_No response_

### Anything else

for sql `insert into test_db.test_table values (7, 'randy', timestamp('2022-07-03 19:11:00'));

the flink connector write data 
image

spark connector  write data 

image

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by hellojinsilei**
## Why are the changes needed?

fix #686 
Use timestamp with zone in iceberg format to solve time zone problems

## How was this patch tested?
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

### CODE REVIEW

File: spark/v3.1/spark/src/main/java/com/netease/arctic/spark/ArcticSparkCatalog.java [hunk 1]
```diff
@@ -206,15 +201,37 @@ public Table createTable(
     }
   }
 
-  private Schema checkAndConvertSchema(Schema schema, Map<String, String> properties) {
+  private Schema checkAndConvertSchema(StructType schema, Map<String, String> properties) {
+    Schema convertSchema;
+    try {
+      CatalogMeta catalogMeta = client.getCatalog(catalog.name());
+      String catalogType = catalogMeta.getCatalogType();
+      if (CATALOG_TYPE_HIVE.equals(catalogType)) {
```
Comment (zhoujinsong) [inline spark/v3.1/spark/src/main/java/com/netease/arctic/spark/ArcticSparkCatalog.java:209]: In my opinion, the hive table use timestamp without zone only, and the iceberg table's timestamp type is dependent on the configuration which is the default with zone.

**Review by zhoujinsong — COMMENTED** [review]
@hellojinsilei  I left some comments. please take another look.

File: site/docs/ch/docker-quickstart.md [hunk 2]
```diff
@@ -126,7 +126,7 @@ CREATE TABLE cdc_source(
 INSERT INTO arctic.test_db.test_table
 SELECT id,
        name,
-       TO_TIMESTAMP(op_time) op_time
+       CAST(TO_TIMESTAMP(op_time) AS TIMESTAMP(6) WITH LOCAL TIME ZONE) op_time
```
Comment (YesOrNo828) [inline site/docs/ch/docker-quickstart.md:129]: Is this field: op_time a timestamp with the time zone in the test_table?
```
create table test_db.test_table(
  id int,
  name string,
  op_time timestamp,
  primary key(id)
) using arctic partitioned by(days(op_time));
```

File: spark/v3.1/spark/src/main/java/com/netease/arctic/spark/ArcticSparkCatalog.java [hunk 3]
```diff
@@ -206,15 +201,39 @@ public Table createTable(
...
 
-  private Schema checkAndConvertSchema(Schema schema, Map<String, String> properties) {
+  private Schema checkAndConvertSchema(StructType schema, Map<String, String> properties) {
+    Schema convertSchema;
+    boolean defaultTimeZoneUsed;
+    try {
+      CatalogMeta catalogMeta = client.getCatalog(catalog.name());
+      String catalogType = catalogMeta.getCatalogType();
+      SparkSession sparkSession = SparkSession.active();
+      if (CATALOG_TYPE_HIVE.equals(catalogType)) {
```
Comment (zhoujinsong) [inline spark/v3.1/spark/src/main/java/com/netease/arctic/spark/ArcticSparkCatalog.java:211]: Can we determine the catalog type by the `ArcticCatalog` implementation? So we may need not the client and rpc to AMS.

File: spark/v3.1/spark/src/main/java/com/netease/arctic/spark/ArcticSparkCatalog.java [hunk 4]
```diff
@@ -206,15 +201,39 @@ public Table createTable(
...
+        defaultTimeZoneUsed = true;
+      } else {
+        defaultTimeZoneUsed = Boolean.parseBoolean(
+            sparkSession.conf().get(USE_TIMESTAMP_WITHOUT_TIME_ZONE_IN_NEW_TABLES,
+                "false"));
```
Comment (zhoujinsong) [inline spark/v3.1/spark/src/main/java/com/netease/arctic/spark/ArcticSparkCatalog.java:216]: Maybe we can change the value in `com.netease.arctic.spark.SparkSQLProperties#USE_TIMESTAMP_WITHOUT_TIME_ZONE_IN_NEW_TABLES_DEFAULT` to `false`?

File: spark/v3.1/spark/src/main/java/com/netease/arctic/spark/ArcticSparkCatalog.java [hunk 5]
```diff
@@ -206,15 +201,39 @@ public Table createTable(
     }
   }
 
-  private Schema checkAndConvertSchema(Schema schema, Map<String, String> properties) {
+  private Schema checkAndConvertSchema(StructType schema, Map<String, String> properties) {
+    Schema convertSchema;
+    boolean defaultTimeZoneUsed;
```
Comment (zhoujinsong) [inline spark/v3.1/spark/src/main/java/com/netease/arctic/spark/ArcticSparkCatalog.java:206]: How about rename this parameter to `useTimestampWithoutZoneInNewTables`?

**Review by zhoujinsong — CHANGES_REQUESTED** [review]
@hellojinsilei  I left some comments. PTAL.

File: site/docs/ch/docker-quickstart.md (hunk 2 above)
Comment (zhoujinsong) [reply site/docs/ch/docker-quickstart.md:129]: Yes, After change in this PR, spark will create timestamp with zone column in test table.

File: spark/v3.1/spark/src/main/java/com/netease/arctic/spark/ArcticSparkCatalog.java (hunk 3 above)
Comment (hellojinsilei) [reply spark/v3.1/spark/src/main/java/com/netease/arctic/spark/ArcticSparkCatalog.java:211]: Does it mean adding a new implementation in ArcticCatalog to determine the catalog type?

Comment (hellojinsilei) [reply spark/v3.1/spark/src/main/java/com/netease/arctic/spark/ArcticSparkCatalog.java:211]: Currently, there is no implementation in ArcticCatalog that can determine the type of catalog

Comment (zhoujinsong) [reply spark/v3.1/spark/src/main/java/com/netease/arctic/spark/ArcticSparkCatalog.java:211]: How about adding a `CatalogUtil#isHiveCatalog` implementation in the hive module?

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Review by baiyangtx — APPROVED** [review]

**Pull request merged**